lib/devise_two_factor/strategies/two_factor_authenticatable.rb

```diff
@@ -21,6 +21,10 @@ def authenticate!
 
       def validate_otp(resource)
         return true unless resource.otp_required_for_login
+
+        # if fingerprints exists fallback to database strategy
+        return true if resource.browser_fingerprint[params['device_fp']].present?
```

Comment:
Should we check here if fingerprint is valid... I mean that not older than 2 days? Or a I missing something?